spec/core/event_spec.rb

```diff
@@ -137,6 +137,12 @@
       it "should not include non-existing fields" do
         expect(subject.include?("doesnotexist")).to be_false
         expect(subject.include?("[j][doesnotexist]")).to be_false
+        expect(subject.include?("[tag][0][hello][yes]")).to be_false
+      end
+
+      it "should include within arrays" do
+        expect(subject.include?("[tags][0]")).to be_true
+        expect(subject.include?("[tags][1]")).to be_false
       end
```

Comment:
I have a test with nesting, up above. or do you specifically mean one with nesting of hash within an array?

```
expect(subject.include?("[c][d]")).to be_true
```